app/jobs/delete_resources_job.rb

```diff
@@ -9,35 +9,14 @@ def perform(url, exhibit)
 
     resp = Faraday.get(url)
 
-    resources = NdjsonNormalizer.normalize(resp.body, url)
-    resources.each do |item|
-      remove_resource(item: item, exhibit: exhibit)
-    end
-
-    logger.info("#{resources.count} records were removed from #{url}.")
-  end
-
-  private
-
-  def remove_resource(item:, exhibit:)
-    resource = DlmeJson.find_by(url: item['id'], exhibit: exhibit)
-    return unless resource
+    resource_hashes = NdjsonNormalizer.normalize(resp.body, url)
+    resource_hashes.each do |resource_hash|
+      resource = DlmeJson.find_by(url: resource_hash['id'], exhibit: exhibit)
+      next unless resource
 
-    # This approach will work for DLME where we only have one exhibit, but
-    # will need to be made more complicated for Spotlight in general because
-    # one resource could be part of more than one exhibit.
-    blacklight_solr.delete_by_query(
-      "{!terms f=#{Spotlight::Engine.config.resource_global_id_field}}#{resource.to_global_id}"
-    )
-    blacklight_solr.commit
-    resource.destroy
-  end
```

Comment:
refactored the rest of the private method functionality into a service class, to re-use in the new rake task